src/EventStore.Core/Telemetry/GCSuspensionMetric.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Diagnostics.Tracing;
+using EventStore.Core.Time;
+
+namespace EventStore.Core.Telemetry;
+
+public class GcSuspensionMetric : EventListener {
+	private const int GcKeyword = 0x0000001;
+	private readonly DurationMaxTracker _tracker;
+	private readonly IClock _clock;
+	private Instant? _started;
+
+	public GcSuspensionMetric(IClock clock, DurationMaxTracker tracker) {
+		_tracker = tracker;
+		_clock = clock;
+	}
+
+	protected override void OnEventSourceCreated(EventSource eventSource) {
+		if (eventSource.Name.Equals("Microsoft-Windows-DotNETRuntime")) {
+			EnableEvents(eventSource, EventLevel.Verbose, (EventKeywords)GcKeyword);
+		}
+	}
+
+	protected override void OnEventWritten(EventWrittenEventArgs eventData) {
+		if (_tracker == null || _clock == null)
+			return;
+
+		if (eventData.EventName is not "GCSuspendEEBegin_V1" && eventData.EventName is not "GCSuspendEEEnd_V1")
+			return;
+
+		switch (eventData.EventName) {
+			case "GCSuspendEEBegin_V1": {
+				var idx = eventData.PayloadNames!.IndexOf("Reason");
+				var value = (uint)eventData.Payload![idx]!;
+
+				// We only track suspensions that are meant for garbage collection.
+				// See https://learn.microsoft.com/en-us/dotnet/fundamentals/diagnostics/runtime-garbage-collection-events
+				if (value == 0x1)
+					_started = _clock.Now;
```

Comment:
i think we'd better use `eventData.TimeStamp` for accuracy

we could add an method to DurationMaxTracker like

```
	public void Record(TimeSpan duration) {
		_recentMax.Record(_clock.Now, duration.TotalSeconds);
	}
```